src/main/java/NestedNullRefactoring.java

```diff
@@ -0,0 +1,151 @@
+
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.List;
+
+import org.eclipse.jdt.core.dom.AST;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.Block;
+import org.eclipse.jdt.core.dom.Expression;
+import org.eclipse.jdt.core.dom.IMethodBinding;
+import org.eclipse.jdt.core.dom.InfixExpression;
+import org.eclipse.jdt.core.dom.MethodDeclaration;
+import org.eclipse.jdt.core.dom.MethodInvocation;
+import org.eclipse.jdt.core.dom.NullLiteral;
+import org.eclipse.jdt.core.dom.ParenthesizedExpression;
+import org.eclipse.jdt.core.dom.PrefixExpression;
+import org.eclipse.jdt.core.dom.PrimitiveType;
+import org.eclipse.jdt.core.dom.ReturnStatement;
+import org.eclipse.jdt.core.dom.SimpleName;
+import org.eclipse.jdt.core.dom.Statement;
+import org.eclipse.jdt.core.dom.Type;
+import org.eclipse.jdt.core.dom.rewrite.ASTRewrite;
+
+public class NestedNullRefactoring extends Refactoring {
+	public static final String NAME = "NestedNullRefactoring";
+
+	private final Dictionary<IMethodBinding, Expression> applicableMethods;
+
+	public NestedNullRefactoring() {
+		applicableMethods = new Hashtable<>();
+	}
+
+	@Override
+	public boolean isApplicable(ASTNode node) {
+
+		if (node instanceof MethodInvocation invocation) {
+			return isApplicable(invocation);
+		}
+
+		// Confirm that the ASTNode is a method
```

Comment:
Does this comment add anything over the code itself? In general, comments should explain _why_ you've taken certain design decisions rather than just repeating information that is readily available from reading the source code. In this case, it's easy to understand the purpose of the instanceof check, so I don't think you need this comment.